### PR TITLE
Fixes issues with closing the branch session in cordova

### PR DESCRIPTION
**Pull request by wedgybo**
This PR fixes the issue with close failing and the queue not processing things properly.

### CODE REVIEW

File: src/0_storage.js [hunk 1]
```diff
@@ -50,16 +50,28 @@ var webStorage = function(perm) {
...
+					allKeyValues[trimPrefix(key)] = retrieveValue(storageMethod.getItem(key));
+				}
 			}
 			return allKeyValues;
 		},
 		get: function(key) { return retrieveValue(storageMethod.getItem(prefix(key))); },
 		set: function(key, value) { storageMethod.setItem(prefix(key), value); },
 		remove: function(key) { storageMethod.removeItem(prefix(key)); },
-		clear: function() { storageMethod.clear(); },
+		clear: function() {
```
Comment (wedgybo) [inline src/0_storage.js]: The clear function for local storage was wiping out everything so I've made it work based on the key prefix

**Comment by dmitrig01** [comment]
hey @wedgybo - this looks great - thanks so much. a few minor comments:

File: src/0_storage.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ goog.require('utils');
 
 var COOKIE_DAYS = 365;
 
-var BRANCH_KEY_PREFIX = 'BRANCH_WEBSDK_KEY';
```
Comment (dmitrig01) [inline src/0_storage.js]: let's keep this - otherwise it'll break old integrations

File: src/3_branch.js [hunk 3]
```diff
@@ -285,6 +285,7 @@ Branch.prototype['init'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done
 			data = setBranchValues(data);
 			if (CORDOVA_BUILD || TITANIUM_BUILD) { // jshint undef:false
 				var first = self._storage.getAll();
+				// I feel this check is wrong?
```
Comment (wedgybo) [inline src/3_branch.js]: The first check still doesn't work as expected. I think there's something up with the logic here.

File: src/0_storage.js (hunk 1 above)
Comment (dmitrig01) [inline src/0_storage.js]: great catch - don't we want to test `item.indexOf(BRANCH_KEY_PREFIX) === 0`?

File: src/0_storage.js (hunk 2 above)
Comment (wedgybo) [inline src/0_storage.js]: Cool. This is the area I can help fix without you guys.

Basically I think you're trying to store two separate keys. One for the first time a session is detected. `branch_data` and the regular `branch_session` so I'm struggling to move forward without understanding what you guys were intending to check for the first session stored in cordova

File: src/0_storage.js (hunk 1 above)
Comment (wedgybo) [inline src/0_storage.js]: Yes, but it's set to something that doesn't match 'branch_session' which is the only key you want to clear when closing the session

File: src/0_queue.js [hunk 4]
```diff
@@ -15,6 +15,6 @@ var Queue = function() {
 	};
 	return function(task) {
 		queue.push(task);
-		if (queue.length == 1) { next(); }
```
Comment (wedgybo) [inline src/0_queue.js]: I don't think this is right solution, but it got it working properly for now.

It looks like the queue never clears, so this condition would stop the close() from running

Comment (dmitrig01) [inline src/0_queue.js]: hmm, i bet there's a case where the callback isn't getting called...

**Pull request merged**